public/app/features/alerting/unified/rule-list/GrafanaRuleLoader.tsx

```diff
@@ -19,52 +20,63 @@ import { RuleActionsButtons } from './components/RuleActionsButtons.V2';
 import { RuleOperation } from './components/RuleListIcon';
 
 const { useGetGrafanaRulerGroupQuery } = alertRuleApi;
+const { useGetGrafanaGroupsQuery } = prometheusApi;
 
 interface GrafanaRuleLoaderProps {
-  rule: GrafanaPromRuleDTO;
-
+  ruleIdentifier: GrafanaRuleIdentifier;
   groupIdentifier: GrafanaRuleGroupIdentifier;
   namespaceName: string;
 }
 
-export function GrafanaRuleLoader({ rule, groupIdentifier, namespaceName }: GrafanaRuleLoaderProps) {
+export function GrafanaRuleLoader({ ruleIdentifier, groupIdentifier, namespaceName }: GrafanaRuleLoaderProps) {
   const {
     data: rulerRuleGroup,
-    isError,
-    isLoading,
+    error: rulerRuleGroupError,
+    isLoading: isRulerRuleGroupLoading,
   } = useGetGrafanaRulerGroupQuery({
     folderUid: groupIdentifier.namespace.uid,
     groupName: groupIdentifier.groupName,
   });
+  const {
+    data: promRuleGroup,
+    error: promRuleGroupError,
+    isLoading: isPromRuleGroupLoading,
+  } = useGetGrafanaGroupsQuery({
+    folderUid: groupIdentifier.namespace.uid,
+    groupName: groupIdentifier.groupName,
+  });
 
-  const rulerRule = rulerRuleGroup?.rules.find((rulerRule) => rulerRule.grafana_alert.uid === rule.uid);
+  const rulerRule = rulerRuleGroup?.rules.find((rulerRule) => rulerRule.grafana_alert.uid === ruleIdentifier.uid);
+  const promRule = promRuleGroup?.data.groups
+    .flatMap((group) => group.rules)
+    .find((promRule) => promRule.uid === ruleIdentifier.uid);
 
-  if (isError) {
-    return <RulerRuleLoadingError rule={rule} />;
+  if (rulerRuleGroupError || promRuleGroupError) {
+    return <RulerRuleLoadingError ruleIdentifier={ruleIdentifier} error={rulerRuleGroupError || promRuleGroupError} />;
   }
 
-  if (isLoading) {
+  if (isRulerRuleGroupLoading || isPromRuleGroupLoading) {
     return <AlertRuleListItemSkeleton />;
   }
 
   if (!rulerRule) {
     return (
       <Alert
-        title={t('alerting.rule-list.cannot-load-rule-details-for', 'Cannot load rule details for {{name}}', {
-          name: rule.name,
+        title={t('alerting.rule-list.cannot-load-rule-details-for', 'Cannot load rule details for UID {{uid}}', {
+          uid: ruleIdentifier.uid,
         })}
         severity="error"
       >
         <Trans i18nKey="alerting.rule-list.cannot-find-rule-details-for">
-          Cannot find rule details for {{ uid: rule.uid ?? '<empty uid>' }}
+          Cannot find rule details for UID {{ uid: ruleIdentifier.uid ?? '<empty uid>' }}
         </Trans>
       </Alert>
     );
   }
 
   return (
     <GrafanaRuleListItem
-      rule={rule}
+      rule={promRule}
```

Comment:
I know this is not related to this pr, but why we don't rename this prop to `promRule`?